src/knx/data_link_layer.cpp

```diff
@@ -49,12 +54,18 @@ void DataLinkLayer::frameRecieved(CemiFrame& frame)
     Priority priority = frame.priority();
     NPDU& npdu = frame.npdu();
     uint16_t ownAddr = _deviceObject.induvidualAddress();
+    SystemBroadcast systemBroadcast = frame.systemBroadcast();
     
     if (source == ownAddr)
         _deviceObject.induvidualAddressDuplication(true);
 
     if (addrType == GroupAddress && destination == 0)
-        _networkLayer.systemBroadcastIndication(ack, type, npdu, priority, source);
+    {
+        if (systemBroadcast == SysBroadcast)
+            _networkLayer.systemBroadcastIndication(ack, type, npdu, priority, source);
+        else 
+            _networkLayer.dataIndication(ack, addrType, destination, type, npdu, priority, source);
```

Comment:
shouldn't this be be broadcatIndication from p. 9 of 3/3/3?

Reply:
Yes. But as you have already pointed out the primitives were missing. 
I can add them accordingly. It is cleaner then.

Reply:
Please do so.

Reply:
done.